### PR TITLE
Int2019 64 create notifications view

### PR DESCRIPTION
**Pull request by robertpin**

### CODE REVIEW

File: ClientApp/viking-vault/src/Common/SideBar.tsx [hunk 1]
```diff
@@ -2,12 +2,16 @@ import React from 'react'
 import './styles.css'
 import dashboard from '../Resources/images/dashboard.png'
 import arrow from '../Resources/images/arrowGRIdeschis.png'
-import { Link } from 'react-router-dom';
 import { UserSideBar } from '../UserDashboard/UserSideBar';
 import { AdminSideBar } from '../AdminDashboard/AdminSideBar';
+import { constants } from '../Resources/Constants';
+import { INotification } from './../UserDashboard/Notifications';
+
+const url = constants.baseUrl+"notifications";
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/Common/SideBar.tsx]: You could rename it to notificationsUrl

File: ClientApp/viking-vault/src/UserDashboard/Notifications.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    id: number;
+    text: string;
+    read: boolean;
+}
+
+interface INotificationState {
+    notifications: INotification[];
+}
+
+const url = constants.baseUrl+"notifications";
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/Notifications.tsx]: You could rename it to notificationsUrl

File: Server/VikingVault.Services/NotificationsService.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                var currentNotification = _dbContext.Notifications
+                    .Include(n => n.User)
+                    .SingleOrDefault(n => n.Id == notification.Id);
+                currentNotification.Read = notification.Read;
+                _dbContext.Notifications.Update(currentNotification);
+                _dbContext.SaveChanges();
+            }
+            catch(Exception e)
+            {
+                throw new NotificationsServiceException();
```
Comment (crisfsabau) [inline Server/VikingVault.Services/NotificationsService.cs]: add also the message from the exception

File: ClientApp/viking-vault/src/UserDashboard/Notifications.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    componentDidMount() {
+        this.getAllNotifications();
+    }
+
+    updateNotificationFrontend = (notification: INotification) => {
+        let allNotifications = this.state.notifications;
+        let notificationToUpdate = allNotifications.find(n => n.id == notification.id);
+        if(notificationToUpdate === undefined)
```
Comment (Slitthe) [inline ClientApp/viking-vault/src/UserDashboard/Notifications.tsx]: Please please please use use curly braces for statements

**Pull request merged**